egs/wsj/s5/steps/nnet3/tdnn/make_configs.py

```diff
@@ -477,36 +497,44 @@ def MakeConfigs(config_dir, splice_indexes_string,
-                               add_final_sigmoid = add_final_sigmoid,
-                               objective_type = objective_type)
-            if xent_regularize != 0.0:
+            if layerwise_pretrain or i == num_hidden_layers - 1:
                 nodes.AddFinalLayer(config_lines, prev_layer_output, num_targets,
-                                    ng_affine_options = " param-stddev=0 bias-stddev=0 learning-rate-factor={0} ".format(
-                                          0.5 / xent_regularize),
-                                    use_presoftmax_prior_scale = use_presoftmax_prior_scale,
-                                    prior_scale_file = prior_scale_file,
-                                    include_log_softmax = True,
-                                    name_affix = 'xent')
-
-        config_files['{0}/layer{1}.config'.format(config_dir, i+1)] = config_lines
-        config_lines = {'components':[], 'component-nodes':[]}
+                                   use_presoftmax_prior_scale = use_presoftmax_prior_scale,
+                                   prior_scale_file = prior_scale_file,
+                                   include_log_softmax = include_log_softmax,
+                                   add_final_sigmoid = add_final_sigmoid,
+                                   objective_type = objective_type)
+                if xent_regularize != 0.0:
+                    nodes.AddFinalLayer(config_lines, prev_layer_output, num_targets,
+                                        ng_affine_options = " param-stddev=0 bias-stddev=0 learning-rate-factor={0} ".format(
+                                              0.5 / xent_regularize),
+                                        use_presoftmax_prior_scale = use_presoftmax_prior_scale,
+                                        prior_scale_file = prior_scale_file,
+                                        include_log_softmax = True,
+                                        name_affix = 'xent')
+        if layerwise_pretrain: 
+          config_files['{0}/layer{1}.config'.format(config_dir, i+1)] = config_lines
+          config_lines = {'components':[], 'component-nodes':[]}
+        elif i == num_hidden_layers - 1:
+          config_files['{0}/layer1.config'.format(config_dir)] = config_lines
 
     left_context += int(parsed_splice_output['left_context'])
     right_context += int(parsed_splice_output['right_context'])
-
     # write the files used by other scripts like steps/nnet3/get_egs.sh
     f = open(config_dir + "/vars", "w")
     print('model_left_context=' + str(left_context), file=f)
     print('model_right_context=' + str(right_context), file=f)
-    print('num_hidden_layers=' + str(num_hidden_layers), file=f)
+    if layerwise_pretrain:
+      print('num_hidden_layers=' + str(num_hidden_layers), file=f)
+    else:
+      print('num_hidden_layers=' + str(1), file=f)
+      print('num_real_hidden_layers=' + str(num_hidden_layers), file=f)
```

Comment:
In my point of view, 'num_of_layer_configs' and 'num_hidden_layers' sound more natural.